internal/controller/apiextensions/composite/composition_ptf.go

```diff
@@ -337,6 +337,9 @@ func (c *PTFComposer) Compose(ctx context.Context, xr resource.Composite, req Co
 	// Subsequent attempts to update that object will therefore fail. This
 	// should be okay; the caller should keep trying until this is a no-op.
 	ao := mergeOptions(filterPatches(allPatches(state.ComposedResources), patchTypesToXR()...))
+	if err := c.client.Status().Update(ctx, state.Composite); err != nil {
+		return CompositionResult{}, errors.Wrap(err, errApplyXR)
+	}
```

Comment:
Are you sure this works? More specifically, are you sure it doesn't "undo" pending changes to other parts of the XR?

My recollection is that when you do a status update like this, the `spec` and `metadata` of `state.Composite` will be reset to whatever the current state is in the API server. i.e. Any pending state changes will be discarded before we reach `Apply`.

Reply:
Looking into this a little further, is it actually possible to take values from a resource's `status.atProvider` in a desired resource and apply it to the XR?

Reply:
Do you need it from a desired resource specifically, or would an observed resource do?

Either way, it's definitely intended that you can update the XR from the status of a composed resource.

Reply:
Here's an example use case. I would like to create a GCP VPC and x number of subnets. Each  subnet would be represented as an array of objects in my definition. My composition function would loop over this array and create a subnetwork for each item in the array and add the desired resources. How would I access the `status.atProvider` fields of those subnets in order to add some of them to the status fields of my definition?